contracts/Collector.sol

```diff
@@ -68,17 +86,27 @@ contract Collector is ICollector{
     }
 
     modifier validShares(RevenuePartner[] memory _partners){
-        uint totalShares = 0;            
-        for(uint i = 0; i < _partners.length; i++)
+        uint totalShares = 0;
+
+        for(uint i = 0; i < _partners.length; i++){
             totalShares = totalShares + _partners[i].share;
+            require(_partners[i].share > 0, "0 is not a valid share");
+        }
 
         require(totalShares == 100, "total shares must add up to 100%");
-
         _;
     }
 
     modifier onlyOwner(){
-        require(msg.sender == owner, "can only call from owner");
+        require(msg.sender == owner, "only owner can call this");
+        _;
+    }
+
+    modifier noBalanceToShare(){
+        uint balance = token.balanceOf(address(this));
+
+        require(balance < partners.length, "there is balance to share");
+
```

Comment:
I'd like to use this to propose a standard for an order of definitions, as follows:
1. private storage
2. internal storage
3. public storage
4. modifiers
5. constructor
6. private methods
7. internal methods
8. external methods
9. public methods

Reply:
Done. Let's talk with teammates to establish this as a standard.

Reply:
done: https://www.notion.so/iovlabs/order-of-definition-8173bded80204f55a9bf88c3b3779a65